src/datumaro/components/algorithms/hash_key_inference/prune.py

```diff
@@ -160,11 +160,9 @@ def base(self, ratio, num_centers, labels, database_keys, item_list, source):
 
         item_id_list = [item.id.split("/")[-1] for item in item_list]
         centroids = [
-            database_keys[item_id_list.index(i.id.split(":")[-1])]
-            for i in list(center_dict.values())
-            if i
+            database_keys[item_id_list.index(i.id)] for i in list(center_dict.values()) if i
         ]
-        kmeans = KMeans(n_clusters=num_centers, n_init=1, init=centroids, random_state=0)
+        kmeans = KMeans(n_clusters=num_centers, n_init=1, init=np.array(centroids), random_state=0)
```

Comment:
```suggestion
        kmeans = KMeans(n_clusters=num_centers, n_init=1, init=np.stack(centroids, axis=0), random_state=0)
```

Is it trying to stack?

Reply:
It is trying to set `centroids` list as (n, 512) shape array. I fixed it with the part you suggested.